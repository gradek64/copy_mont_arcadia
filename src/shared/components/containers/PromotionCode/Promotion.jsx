import PropTypes from 'prop-types'
import React, { Component } from 'react'

export default class Promotion extends Component {
  static propTypes = {
    promotion: PropTypes.object.isRequired,
    showAddButton: PropTypes.bool,
    removePromotionCode: PropTypes.func,
    showForm: PropTypes.func,
  }

  render() {
    const {
      showAddButton,
      removePromotionCode,
      showForm,
      promotion,
    } = this.props
    const { label, promotionCode } = promotion
    const actionRowClass = showAddButton ? ' PromotionCode-codeActionSpace' : ''
    return (
      <div key={promotionCode} className="PromotionCode-code">
        <span className="PromotionCode-codeTitle"> {promotionCode} </span>
        <p className="PromotionCode-codeDescription">{label}</p>
        <div className={`PromotionCode-codeActions${actionRowClass}`}>
          {showAddButton && (
            <a // eslint-disable-line jsx-a11y/no-static-element-interactions
              className="PromotionCode-addText"
              onClick={showForm}
            >
              {' '}
              Add another
            </a>
          )}
          <a // eslint-disable-line jsx-a11y/no-static-element-interactions
            className="PromotionCode-removeText"
            onClick={() => removePromotionCode(promotionCode)}
          >
            Remove
          </a>
        </div>
      </div>
    )
  }
}
