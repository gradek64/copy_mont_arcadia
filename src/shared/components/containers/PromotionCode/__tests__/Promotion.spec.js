import testComponentHelper from 'test/unit/helpers/test-component'
import Promotion from '../Promotion'

describe('<Promotion />', () => {
  const initialProps = {
    promotion: {
      promotionCode: 'WELCOME15',
      label: 'Welcome treat! 15% off',
    },
    removePromotionCode: jest.fn(),
    showForm: jest.fn(),
  }
  const renderComponent = testComponentHelper(Promotion)
  beforeEach(() => jest.resetAllMocks())
  describe('@renders', () => {
    it('in default state', () => {
      expect(renderComponent(initialProps).getTree()).toMatchSnapshot()
    })
    it('with add Button', () => {
      expect(
        renderComponent({
          ...initialProps,
          showAddButton: true,
        }).getTree()
      ).toMatchSnapshot()
    })
  })
  describe('@events', () => {
    describe('click remove promo', () => {
      it('it should call removePromotionCode() with props.promotion.promotionCode', () => {
        const { wrapper, instance } = renderComponent(initialProps)
        expect(instance.props.removePromotionCode).not.toHaveBeenCalled()
        wrapper.find('.PromotionCode-removeText').simulate('click')
        expect(instance.props.removePromotionCode).toHaveBeenCalledTimes(1)
        expect(instance.props.removePromotionCode).toHaveBeenLastCalledWith(
          instance.props.promotion.promotionCode
        )
      })
    })
    describe('click add promocode', () => {
      it('it should call removePromotionCode() with props.promotion.promotionCode', () => {
        const { wrapper, instance } = renderComponent({
          ...initialProps,
          showAddButton: true,
        })
        expect(instance.props.showForm).not.toHaveBeenCalled()
        wrapper.find('.PromotionCode-addText').simulate('click')
        expect(instance.props.showForm).toHaveBeenCalledTimes(1)
      })
    })
  })
})
