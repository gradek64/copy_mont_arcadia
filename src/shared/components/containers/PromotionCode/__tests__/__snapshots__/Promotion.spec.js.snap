// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Promotion /> @renders in default state 1`] = `
<div
  className="PromotionCode-code"
  key="WELCOME15"
>
  <span
    className="PromotionCode-codeTitle"
  >
     
    WELCOME15
     
  </span>
  <p
    className="PromotionCode-codeDescription"
  >
    Welcome treat! 15% off
  </p>
  <div
    className="PromotionCode-codeActions"
  >
    <a
      className="PromotionCode-removeText"
      onClick={[Function]}
    >
      Remove
    </a>
  </div>
</div>
`;

exports[`<Promotion /> @renders with add Button 1`] = `
<div
  className="PromotionCode-code"
  key="WELCOME15"
>
  <span
    className="PromotionCode-codeTitle"
  >
     
    WELCOME15
     
  </span>
  <p
    className="PromotionCode-codeDescription"
  >
    Welcome treat! 15% off
  </p>
  <div
    className="PromotionCode-codeActions PromotionCode-codeActionSpace"
  >
    <a
      className="PromotionCode-addText"
      onClick={[MockFunction]}
    >
       
      Add another
    </a>
    <a
      className="PromotionCode-removeText"
      onClick={[Function]}
    >
      Remove
    </a>
  </div>
</div>
`;
